crates/ide_assists/src/handlers/extract_module.rs

```diff
@@ -779,7 +779,12 @@ fn get_replacements_for_visibilty_change(
             ast::Item::Enum(it) => replacements.push((it.visibility(), it.syntax().clone())),
             ast::Item::ExternCrate(it) => replacements.push((it.visibility(), it.syntax().clone())),
             ast::Item::Fn(it) => replacements.push((it.visibility(), it.syntax().clone())),
-            ast::Item::Impl(it) => impls.push(it),
+            ast::Item::Impl(it) => {
+                //Associated item's visibility should not be changed
+                if let None = it.for_token() {
+                    impls.push(it);
+                }
+            }
```

Comment:
```suggestion
            //Associated item's visibility should not be changed
            ast::Item::Impl(it) if it.for_token().is_none() => impls.push(it),
```
Can be written like this a bit more succinct